pandas/core/strings.py

```diff
@@ -2114,19 +2114,55 @@ def rjust(self, width, fillchar=' '):
 
     def zfill(self, width):
         """
-        Filling left side of strings in the Series/Index with 0.
-        Equivalent to :meth:`str.zfill`.
+        Pad strings in the Series/Index by prepending '0' characters.
+
+        Strings in the Series/Index are padded with prepending '0' characeters
+        (i.e. on the left of the string) to reach a total string length
+        of `width`.  in the Series/Index with length greater than `width`
+        are unchanged.
+
+        Note: Differs from :meth:`str.zfill` which has special handling
+        for '+'/'-' in the string.
```

Comment:
There is a `Notes` section in the numpy docstring convention that we can use for this:

```
Notes
-------
```

https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-6-notes